### ISSUE TITLE
worker scope onmessage/MessageEvent bad typings

### ISSUE THREAD

**Issue by crowlKats** [denoland/deno#7839]
`self.onmessage = async (msg: MessageEvent) => {}` wont work because 
```
error: Uncaught Error: TS2322 [ERROR]: Type '(msg: MessageEvent) => Promise<void>' is not assignable to type '((e: MessageEvent) => void) & ((e: { data: any; }) => void | Promise<void>)'.
  Type '(msg: MessageEvent) => Promise<void>' is not assignable to type '(e: { data: any; }) => void | Promise<void>'.
    Types of parameters 'msg' and 'e' are incompatible.
      Type '{ data: any; }' is missing the following properties from type 'MessageEvent': origin, lastEventId, bubbles, cancelBubble, and 17 more.
```

and `self.onmessage = async (msg) => {}` wont work because `Parameter 'msg' implicitly has an 'any' type.`
It seems to be happening since 1.4.3.

from the `((e: { data: any; }) => void | Promise<void>)` i can see that `MessageEvent.data` was wanted to be of type `any`, but  from the typescript typings `MessageEvent` should actually take a type param that defines the type of `MessageEvent.data`

**Comment by 00ff0000red** [comment]
I believe that the `addEventListener` method's types are also broken, but I can't confirm this myself at the moment.

**Comment by kitsonk** [comment]
@00ff0000red wasn't that fixed in #7791?  If it is still broken, please open another issue.
